### ISSUE TITLE
Fallback routes not working with nested components

### ISSUE THREAD

**Issue by lpellegr** [vaadin/vaadin-router#284]
Let's say you have the following routes:

```
router.setRoutes([
    {path: '/terms', component: 'x-terms'}
    {path: '/service',
      component: 'x-navigation-layout',
      children: [
        {path: '/', component: 'x-home-view'},
        {path: '/users', component: 'x-user-list'},
        {path: '/users/:user', component: 'x-user-profile'},
      ]
    },
    {path: '(.*)', component: 'x-page-not-found'}
  ]);
```

- If you move to `/terms`, component `x-terms` is rendered.
- If you move to `/termsTralala`, component `x-page-not-found` is rendered.
- If you move to `/service/users`, component `x-user-list` is rendered.
- If you move to `/service/usersTralala`, an error is thrown:

> Error: [Vaadin.Router] Page not found (/service/usersTralala)
>     at getNotFoundError (vaadin-router.js:135)
>     at amendedContext.next.then.nextContext (vaadin-router.js:1585)

I would expected the parent fallback route to match and 'x-page-not-found' to be rendered but it's not the case.

In other words, all parent routes are not tried when a nested route is not completely matched. It looks like a bug.

**Comment by lpellegr** [comment]
Any update?

**Comment by vlukashov** [comment]
This bug is in the backlog and I would like to see it fixed eventually. But no specific time line can be given at this point.

If this issue becomes a blocker there is a workaround: include the `(.*)` route as the last child in every list of nested routes. It's not pretty but it can be used until there is a better fix.

```js
router.setRoutes([
    {path: '/terms', component: 'x-terms'}
    {path: '/service',
      component: 'x-navigation-layout',
      children: [
        {path: '/', component: 'x-home-view'},
        {path: '/users', component: 'x-user-list'},
        {path: '/users/:user', component: 'x-user-profile'},
        {path: '(.*)', component: 'x-page-not-found'}
      ]
    },
    {path: '(.*)', component: 'x-page-not-found'}
  ]);
```

**Comment by SilvanTheuma** [comment]
Any update on this? I have the same issue. Also when adding the `(.*)` to the children loads the `x-page-not-found` component immediately.

**Comment by kuka-radovan** [comment]
Is this issue planned to fix? I am facing the same problem because I don't want to have inner 404 screen to be wrapped by layout component.

**Comment by haijian-vaadin** [comment]
Hi, this issue is now in our sprint backlog, we will share more details once we start working on it.

**Comment by haijian-vaadin** [comment]
@platosha was too fast, we ended up closing this ticket earlier than expected. The fix is now in [v1.7.0](https://github.com/vaadin/vaadin-router/releases)

**Comment by VitoMedlej** [comment]
is this fixed or not? im getting an error from a nested components as GET http://localhost:8080/gaming/main.js net::ERR_ABORTED 404 (Not Found) on reload ,everything else works fine except a nested url .anyone knows how to fix that?
